File: retrocast-client/Retrocast/Sources/Models/Member.swift
import Foundation

struct Member: Codable, Identifiable, Hashable, Sendable {
    let guildID: Snowflake
    let userID: Snowflake
    var nickname: String?
    let joinedAt: Date
    var roles: [Snowflake]

    var id: Snowflake { userID }

    enum CodingKeys: String, CodingKey {
        case guildID = "guild_id"
        case userID = "user_id"
        case nickname
        case joinedAt = "joined_at"
        case roles
    }
}

File: retrocast/internal/models/member.go
package models

import "time"

type Member struct {
	GuildID  int64    `json:"guild_id,string"`
	UserID   int64    `json:"user_id,string"`
	Nickname *string  `json:"nickname,omitempty"`
	JoinedAt time.Time `json:"joined_at"`
	Roles    []int64  `json:"roles"`
}
